services/maintenance/maintenance.service.js

```diff
@@ -1,27 +1,56 @@
 'use strict'
 
-const LegacyService = require('../legacy-service')
-const { makeBadgeData: getBadgeData } = require('../../lib/badge-data')
-const log = require('../../core/server/log')
-
-// This legacy service should be rewritten to use e.g. BaseJsonService.
-//
-// Tips for rewriting:
-// https://github.com/badges/shields/blob/master/doc/rewriting-services.md
-//
-// Do not base new services on this code.
-module.exports = class Maintenance extends LegacyService {
-  static get category() {
-    return 'other'
-  }
+const { BaseService } = require('..')
 
+module.exports = class Maintenance extends BaseService {
```

Comment:
Probably this should use `NonMemoryCachingBaseService` as there's nothing being fetched here.